strawberryfields/tdm/tdmprogram.py

```diff
@@ -350,6 +382,10 @@ def construct_circuit(self):
             elif isinstance(self.shift, int):
                 q = shift_by(q, self.shift)  # shift at end of each time bin
 
+        self.unrolled_circuit = self.circuit.copy()
```

Comment:
Took some time for me to understand how the `copy` calls and setting the refs worked between `self.circuit`, `self.rolled_circuit` and `self.unrolled_circuit`. Maybe worth placing a couple of comments? (Similar to this suggestion or something more descriptive)

Also: could there be a case where `self.circuit` does not equal neither `self.rolled_circuit` nor `self.unrolled_circuit`?
```suggestion
        # Unrolling the circuit for the first time: storing a copy of the unrolled circuit
        self.unrolled_circuit = self.circuit.copy()
```

Reply:
Good suggestion @antalszava.

> Also: could there be a case where self.circuit does not equal neither self.rolled_circuit nor self.unrolled_circuit?

There shouldn't be, no. If there is, then there is a bug in the logic.